app/code/Magento/Quote/Model/QuoteManagement.php

```diff
@@ -532,19 +532,28 @@ protected function submitQuote(QuoteEntity $quote, $orderData = [])
             );
             $this->quoteRepository->save($quote);
         } catch (\Exception $e) {
-            if (!empty($this->addressesToSync)) {
-                foreach ($this->addressesToSync as $addressId) {
-                    $this->addressRepository->deleteById($addressId);
+            try {
+                if (!empty($this->addressesToSync)) {
+                    foreach ($this->addressesToSync as $addressId) {
+                        $this->addressRepository->deleteById($addressId);
+                    }
                 }
+                $this->eventManager->dispatch(
+                    'sales_model_service_quote_submit_failure',
+                    [
+                        'order' => $order,
+                        'quote' => $quote,
+                        'exception' => $e,
+                    ]
+                );
+            } catch (\Exception $consecutiveException) {
+                $message = sprintf(
+                    "An exception occurred on 'sales_model_service_quote_submit_failure' event: %s",
+                    $consecutiveException->getMessage()
+                );
+
+                throw new \Exception($message, 0, $e);
```

Comment:
I just want to be sure that we don't have any exception missing. We don't throw the `$e` directly if a consecutive exception appears but pass  `$e` as a `previous` argument upon the consecutive exception throwing. Have you double checked that the original exception is being shown in logs even when it's not thrown directly?

Reply:
I checked it before - see the _Example_ section in pull request description. As you suggested I double checked it:

The exception thrown in this line must not be a `\Magento\Framework\Exception\LocalizedException` as it would bypass the logger and cause the exception message to be passed to store front in `\Magento\Checkout\Model\GuestPaymentInformationManagement::savePaymentInformationAndPlaceOrder` (line 110ff).
The `\Exception` is passed to the `\Magento\Framework\Logger\Monolog::addRecord` in `\Magento\Checkout\Model\GuestPaymentInformationManagement::savePaymentInformationAndPlaceOrder` (line 116). The Magento Logger uses the `\Monolog\Formatter\LineFormatter`. In its method `convertToString` the exception is encoded as json - and thus the consecutive exception is contained in the log message.